src/_nebari/stages/infrastructure/__init__.py

```diff
@@ -118,6 +118,7 @@ class AzureInputVars(schema.Base):
 class AWSNodeGroupInputVars(schema.Base):
     name: str
     instance_type: str
+    custom_ami: Optional[str] = None
```

Comment:
to follow the aws standard naming, let's use `ami_type`

Reply:
hello @viniciusdc,
in this case, the `custom_ami` prompts the user to input an AMI Id, as opposed to the ami_type (e.g. "CUSTOM").  The corresponding aws var here is `image_id`.  My impression of `image_id` is that it seems a bit ambiguous for nebari config purposes since it does not make it obvious that an ami is expected.

Reply:
@viniciusdc how about using `ami_image_id` ?

Reply:
@joneszc I see; I was under the impression `ami_type` would support the ID of a custom AMI as well, but after your comment and re-reding the docs, that does not seem to be true (unfortunately). I agree with you that image_id would also be ambiguous, considering the overall shape of the nebari-config. @tylergraff `ami_image_id` would be nice, but after check-in what both settings do I agree with @joneszc original suggestion as it will resemble the fact that this is a "CUSTOM" [AMItype](https://docs.aws.amazon.com/eks/latest/APIReference/API_Nodegroup.html#AmazonEKS-Type-Nodegroup-amiType)

Reply:
We could also have something like this:
```python
class AWSAMI(BaseModel):
    type: Literal["CUSTOM", None] = None
    id: Optional[str] = Field(default=None, description="AMI ID. Required if type is set to CUSTOM")

    @RootValidator
    def check_custom_ami_id(cls, values):
        ami_type = values.get('type')
        ami_id = values.get('id')
        if ami_type == "CUSTOM" and not ami_id:
            raise ValueError('AMI ID must be provided when type is CUSTOM.')
        return values

class AWSNodeGroupInputVars(BaseModel):
    name: str
    instance_type: str
    ami: Optional[AWSAMI] = None
```
the benefit would be that we can later extend this to include the `AL2_x86_64_GPU/AL2_x86_64` to clean the conditionals logic in the provider itself.